nodeClient.go

```diff
@@ -523,6 +525,94 @@ func (rc *NodeClient) SubmitTransaction(signedTxn *SignedTransaction) (data *api
 	return
 }
 
+type EstimateGasUnitPrice bool
+
+type EstimateMaxGasAmount bool
+
+type EstimatePrioritizedGasUnitPrice bool
+
+func (rc *NodeClient) SimulateTransaction(rawTxn *RawTransaction, sender TransactionSigner, options ...any) (data *[]api.UserTransaction, err error) {
+	// build authenticator for simulation
+	var auth *crypto.AccountAuthenticator
+	derivationScheme := sender.PubKey().Scheme()
+	switch derivationScheme {
+	case crypto.Ed25519Scheme:
+		auth = &crypto.AccountAuthenticator{
+			Variant: crypto.AccountAuthenticatorEd25519,
+			Auth: &crypto.Ed25519Authenticator{
+				PubKey: sender.PubKey().(*crypto.Ed25519PublicKey),
+				Sig:    &crypto.Ed25519Signature{Inner: [64]byte(make([]byte, 64))},
+			},
+		}
+	case crypto.SingleKeyScheme:
+		mockSig := &crypto.AnySignature{}
+		_ = mockSig.FromBytes(make([]byte, 64))
+		auth = &crypto.AccountAuthenticator{
+			Variant: crypto.AccountAuthenticatorSingleSender,
+			Auth: &crypto.SingleKeyAuthenticator{
+				PubKey: sender.PubKey().(*crypto.AnyPublicKey),
+				Sig:    mockSig,
+			},
+		}
+	case crypto.MultiEd25519Scheme:
+	case crypto.MultiKeyScheme:
+		// todo: add support for multikey simulation
+		return nil, fmt.Errorf("currently unsupported sender derivation scheme %v", derivationScheme)
+	default:
```

Comment:
Interesting, will probably refactor this later to just have a function `pubkey.ToSimulateAuthenticator()` or something like that.  Either that or a function `signForSimulation`